services/wfm/src/main/java/org/openkilda/wfm/topology/flow/bolts/CrudBolt.java

```diff
@@ -309,26 +305,35 @@ private void handleDeleteRequest(String flowId, CommandMessage message, Tuple tu
     private void handleCreateRequest(CommandMessage message, Tuple tuple) throws IOException {
         Flow requestedFlow = ((FlowCreateRequest) message.getData()).getPayload();
 
+        ImmutablePair<PathInfoData, PathInfoData> path;
         try {
-            ImmutablePair<PathInfoData, PathInfoData> path = pathComputer.getPath(requestedFlow, Strategy.HOPS);
-            logger.info("Created flow path: {}", path);
-
-            ImmutablePair<Flow, Flow> flow = flowCache.createFlow(requestedFlow, path);
-            logger.info("Created flow: {}", flow);
-
-            FlowInfoData data = new FlowInfoData(requestedFlow.getFlowId(), flow, FlowOperation.CREATE,
-                    message.getCorrelationId());
-            InfoMessage infoMessage = new InfoMessage(data, System.currentTimeMillis(), message.getCorrelationId());
-            Values topology = new Values(MAPPER.writeValueAsString(infoMessage));
-            outputCollector.emit(StreamType.CREATE.toString(), tuple, topology);
-
-            Values northbound = new Values(new InfoMessage(new FlowResponse(buildFlowResponse(flow)),
-                    message.getTimestamp(), message.getCorrelationId(), Destination.NORTHBOUND));
-            outputCollector.emit(StreamType.RESPONSE.toString(), tuple, northbound);
+            path = pathComputer.getPath(requestedFlow, Strategy.HOPS);
         } catch (UnroutablePathException e) {
             throw new MessageException(message.getCorrelationId(), System.currentTimeMillis(),
                     ErrorType.CREATION_FAILURE, "Could not create flow", "Path was not found");
         }
+
+        logger.info("Created flow path: {}", path);
+
+        try {
+            new FlowValidator(flowCache).checkFlowForEndpointConflicts(requestedFlow);
```

Comment:
Why not do it before pathComputer.getPath()? It is obvious that getPath is more "expensive".

PS Why not to place both this calls in one try/catch block?

Reply:
agreed, seems to make more sense to do before the call to getPath

Reply:
Make sense. I'll change the order of calls.